### DISCUSSION TITLE
Imilab V1 Room Cleaning

### DISCUSSION THREAD

**Discussion by Zsola576** [merdok/homebridge-miot#492] [Q&A]
Hello!

I hope somebody can help me about the Imilab V1 Room Cleaning.
Sorry, i'm a bigginer in this theme.
I tried to make action buttons for each room, but if i hit this button, the robot going to clean 3 different room, i not really understand why.

Miot didn't give me the roomID-s, whatever i did...

Here is the program:

                {
                    "name": "Consuela",
                    "ip": "xxxxxxxxx",
                    "token": "xxxxxxxxxx",
                    "deviceId": "xxxxxxxx",
                    "model": "chuangmi.vacuum.hmi707",
                    "pollingInterval": 10,
                    "deepDebugLog": false,
                    "buzzerControl": true,
                    "ledControl": true,
                    "childLockControl": true,
                    "modeControl": true,
                    "actionButtons": [
                        {
                            "action": "vacuum:start-room-sweep",
                            "name": "Clean kitchen",
                            "params": [
                                "{\"selects\":[[3,1,1,2,1]]}",
                                {
                                    "piid": 1,
                                    "value": 18
                                }
                            ]
                        }
                    ]
                }

These are the vacuum services, properties, and actions from homebridge miot:

[2023. 07. 04. 9:07:59] [miot] [Consuela] Device services: [
  "vacuum",
  "battery",
  "speaker",
  "filter",
  "brush-cleaner",
  "brush-cleaner",
  "brush-cleaner",
  "custom",
  "map",
  "sweep"
]
[2023. 07. 04. 9:07:59] [miot] [Consuela] Initializing device properties
[2023. 07. 04. 9:07:59] [miot] [Consuela] Device properties: [
  "vacuum:status",
  "vacuum:fault",
  "vacuum:on",
  "vacuum:mode",
  "vacuum:sweep-type",
  "vacuum:room-ids",
  "battery:battery-level",
  "battery:charging-state",
  "speaker:volume",
  "speaker:mute",
  "filter:filter-life-level",
  "filter:filter-left-time",
  "brush-cleaner:brush-left-time",
  "brush-cleaner:brush-life-level",
  "brush-cleaner11:brush-left-time",
  "brush-cleaner11:brush-life-level",
  "brush-cleaner12:brush-left-time",
  "brush-cleaner12:brush-life-level",
  "custom:mop",
  "custom:work-station-frep",
  "custom:timing",
  "custom:clean-area",
  "custom:uid",
  "custom:auto-boost",
  "custom:forbid-time",
  "custom:water-level",
  "custom:total-clean-time",
  "custom:total-clean-areas",
  "custom:clean-counts",
  "custom:clean-time",
  "custom:double-clean",
  "custom:edge-sweep",
  "custom:led-switch",
  "custom:lidar-collision",
  "custom:station-key",
  "custom:station-key-led",
  "custom:current-audio",
  "custom:audio",
  "custom:audio-progress",
  "custom:station-type",
  "custom:voice-config",
  "map:clean-path",
  "map:restricted-zone",
  "map:auto-area",
  "map:map-memory",
  "map:map-name",
  "map:use-auto-area",
  "map:path-type",
  "sweep:sweep-mode",
  "sweep:clean-info"
]
[2023. 07. 04. 9:07:59] [miot] [Consuela] Initializing device actions
[2023. 07. 04. 9:07:59] [miot] [Consuela] Device actions: [
  "vacuum:start-sweep",
  "vacuum:stop-sweeping",
  "vacuum:start-room-sweep",
  "battery:start-charge",
  "filter:reset-filter-life",
  "brush-cleaner:reset-brush-life",
  "brush-cleaner11:reset-brush-life",
  "brush-cleaner12:reset-brush-life",
  "custom:find-robot",
  "custom:stop-find-charge",
  "custom:continue-sweep",
  "custom:start-dust",
  "custom:pause",
  "custom:pause-find-charge",
  "custom:continue-find-charge",
  "custom:set-audio",
  "custom:get-audio-conf",
  "map:request-path",
  "map:change-map-name",
  "map:aet-auto-area",
  "map:delete-map-list",
  "map:reset-current-map",
  "map:set-current-map",
  "map:change-map-name7",
  "map:custom-area",
  "map:area-order",
  "sweep:start-sweep"
]

Thanks in advance for your help!

**Comment by merdok** [comment]
Hi, 
your device seems new, at least to me, so you would probably need to analyse what exactly protocol it follows for the room cleaning. That is usually not documented and you can either do some try and error or try to capture the packets which is he mi home app sends and analyse them.

But I guess if it already cleaned 3 rooms when you hit the button, then I guess you would need to adjust the parameters of the command to reduce the number of rooms or select a different area, so you might be close.
